### PR TITLE
RavenDB-15744 Support logging toggling

### PR DESCRIPTION
**Pull request by haludi**
Closed PR to v5.2
https://github.com/ravendb/ravendb/pull/14790

### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-15744/Support-logging-toggling

### Additional description
* The switch loggers have a hierarchy, parent - children. (the current structure is present in "Switch loggers structure")
* Setting switch logger will not affect its already existing children
* A new child will be set as its parent 
* Setup the Logging source will affect only "reset" loggers.

Switch loggers structure:
```
 Generic: 
  Name: "Generic"
  Loggers" 
   - Name: "Memory"
 Server: 
  - Name: "Server"
   Loggers: 
   - Name: "Databases"
    Loggers: 
     - Name: "DB1"
      Loggers: 
       - Name: "Indexes"
         Loggers": 
          - Name: "Index1"
          - Name: "Index2"
     - Name: "DB2"
   - Name: "Cluster"
```
### Type of change
- New feature

### How risky is the change?
- High

### Backward compatibility
- Non-breaking change

### Is it a platform-specific issue?
- No

### Documentation update
- No documentation update is needed 

### Testing 
- Tests have been added that prove the fix is effective or that the feature works

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- UI work can be added

### CODE REVIEW

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 1]
```diff
@@ -237,5 +245,105 @@ public Task DisableMicrosoftLog()
...
+            generic = genericLogger;
+            server = serverLogger;
+            return new DisposableAction(() =>
+            {
+                Monitor.Exit(genericLogger);
+                Monitor.Exit(serverLogger);
+            });
+        }
+
+        [RavenAction("/admin/logging-toggling/loggers", "GET", AuthorizationStatus.Operator)]
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: `/admin/logging-toggling/loggers` => `/admin/logs/loggers`

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 2]
```diff
@@ -237,5 +245,105 @@ public Task DisableMicrosoftLog()
...
+                {
+                    ["LogMode"] = LoggingSource.Instance.LogAvailability.GetMode(),
+                    ["Loggers"] = new DynamicJsonValue {[generic.Name] = generic.ToJson(), [server.Name] = server.ToJson()}
+                };
+                var json = context.ReadObject(djv, "logs/loggers");
+                writer.WriteObject(json);
+            }
+        }
+        
+        [RavenAction("/admin/logging-toggling/configuration", "GET", AuthorizationStatus.Operator)]
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: `/admin/logs/loggers/configuration`

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 3]
```diff
@@ -237,5 +245,105 @@ public Task DisableMicrosoftLog()
...
+                SwitchLoggerConfigurationHelper.GetConfigurationFromRoot(generic, configuration);
+                SwitchLoggerConfigurationHelper.GetConfigurationFromRoot(server, configuration);
+
+                var djv = new DynamicJsonValue {[nameof(SwitchLoggerConfiguration.Loggers)] = configuration.ToJson()};
+                var json = context.ReadObject(djv, "loggers/configuration");
+                writer.WriteObject(json);
+            }
+        }
+        
+        [RavenAction("/admin/logging-toggling/configuration", "POST", AuthorizationStatus.Operator)]
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: `/admin/logs/loggers/configuration`

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs [hunk 4]
```diff
@@ -237,5 +245,105 @@ public Task DisableMicrosoftLog()
...
+        public async Task SetLoggerMode()
+        {
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext ctx))
+            {
+                var input = await ctx.ReadForMemoryAsync(RequestBodyStream(), "SwitchLoggerConfiguration");
+                if (input.TryGet("Configuration", out BlittableJsonReaderObject json) == false)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: I think we can drop `Configuraton` then GET and POST would have same format? Thoughts?

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 4 above)
Comment (haludi) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Done

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 3 above)
Comment (haludi) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Done

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 2 above)
Comment (haludi) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Done

File: src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs (hunk 1 above)
Comment (haludi) [reply src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs]: Done

**Pull request closed without merging**